ui-spacetimechart/src/components/workSchedules/Region.tsx

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useDraw } from '../../hooks/useCanvas';
+import { type DrawingFunction } from '../../lib/types';
+import { type SpaceTimeRegion } from '../../types';
+
+const Region = ({
+  timeStart, // x date timestamp
+  timeEnd, // x date timestamp
+  spaceStart, // y metre
+  spaceEnd, // y metre
+  image,
+}: SpaceTimeRegion) => {
+  const [imageElement, setImageElement] = useState<HTMLImageElement | null>(null);
+  useEffect(() => {
+    if (image) {
+      const newImage = new Image();
+      newImage.src = image;
+      newImage.onload = () => {
+        setImageElement(newImage);
+      };
+    }
+  }, [image]);
+  const drawRegion = useCallback<DrawingFunction>(
+    (ctx, { getSpacePixel, getTimePixel, spaceAxis }) => {
+      if (!imageElement) {
+        return;
+      }
+
+      const timeStartPixel = getTimePixel(timeStart);
+      const endTimePixel = getTimePixel(timeEnd);
+      const spaceStartPixel = getSpacePixel(spaceStart);
+      const spaceEndPixel = getSpacePixel(spaceEnd);
+
+      const areaSpaceSize = spaceEndPixel - spaceStartPixel;
+      const areaTimeSize = endTimePixel - timeStartPixel;
+      if (!areaSpaceSize || !areaTimeSize) return;
+
+      const pattern = ctx.createPattern(imageElement, 'repeat');
+      if (!pattern) {
+        return;
+      }
+
+      ctx.save();
+      ctx.fillStyle = pattern;
+      if (spaceAxis === 'x') {
+        ctx.translate(spaceStartPixel, timeStartPixel);
```

Comment:
Is the `translate()` call necessary? Could we just pass the x/y as the first argument to `fillRect()`?

Reply:
It’s necessary, we want the upper left corner of the svg to be 0,0. otherwise the svg dashes move with the chart when we scroll the chart horizontally. I think it has to do with the way ctx.createPattern works.

Reply:
WFM!